homeassistant/components/senseme/config_flow.py

```diff
@@ -28,6 +29,21 @@ def __init__(self) -> None:
         self._discovered_devices: list[SensemeDevice] | None = None
         self._discovered_device: SensemeDevice | None = None
 
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle discovery via dhcp."""
+        # If discovery is already running, it takes precedence since its more efficient
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        if (
+            not (device := await async_get_device_by_ip_address(discovery_info.ip))
+            or not device.uuid
+        ):
+            return self.async_abort(reason="cannot_connect")
+        device.stop()
```

Comment:
If there is a `device`, but the `uuid` is None, we still would need to call `stop()` right?

Reply:
Handled in 12ab4c8b4fb84d791b7d9aa07760de75f25cf17e